app/src/main/java/com/example/rocketreserver/Apollo.kt

```diff
@@ -15,21 +15,20 @@ fun apolloClient(context: Context): ApolloClient {
         "Only the main thread can get the apolloClient instance"
     }
 
-    if (instance != null) {
-        return instance!!
-    }
-
     val okHttpClient = OkHttpClient.Builder()
         .addInterceptor(AuthorizationInterceptor(context))
         .build()
 
-    instance = ApolloClient.builder()
+    return instance ?: ApolloClient.builder()
         .serverUrl("https://apollo-fullstack-tutorial.herokuapp.com/graphql")
-        .subscriptionTransportFactory(WebSocketSubscriptionTransport.Factory("wss://apollo-fullstack-tutorial.herokuapp.com/graphql", okHttpClient))
+        .subscriptionTransportFactory(
+            WebSocketSubscriptionTransport.Factory(
+                "wss://apollo-fullstack-tutorial.herokuapp.com/graphql",
+                okHttpClient
+            )
+        )
         .okHttpClient(okHttpClient)
-        .build()
-
-    return instance!!
+        .build().also { instance = it }
```

Comment:
I don't want to bikeshed this too much but in this specific case, I think there's some value in keeping the formatting as close as possible to what's in the documentation (https://www.apollographql.com/docs/android/tutorial/10-authenticate-your-queries/#use-the-interceptor). Since this is a tutorial aimed at first timers, we want as little friction as possible. Do you mind keeping the same formating there? Or updating the matching documentation at https://github.com/apollographql/apollo-android ?

Reply:
Keeping the documentation formatting is better choice.